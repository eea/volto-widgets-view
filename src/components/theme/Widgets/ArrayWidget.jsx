import React from 'react';
import cx from 'classnames';

export const ArrayWidget = ({ children, className }) => (
  <span className={cx(className, 'array', 'widget')}>
    {children
      ? children.map((child) => (
          <span key={child.token || child.title || child}>
            {child.title || child.token || child}
          </span>
        ))
      : ''}
  </span>
);
