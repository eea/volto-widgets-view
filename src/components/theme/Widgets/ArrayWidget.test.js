import React from 'react';
import renderer from 'react-test-renderer';
import { ArrayWidget } from './ArrayWidget';

describe('ArrayWidget', () => {
  it('renders a simple array view widget component', () => {
    const component = renderer.create(
      <ArrayWidget className="metadata">{['foo', 'bar']}</ArrayWidget>,
    );
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });

  it('renders a vocabulary array view widget component', () => {
    const component = renderer.create(
      <ArrayWidget className="metadata">
        {[{ title: 'Foo' }, { title: 'Bar' }]}
      </ArrayWidget>,
    );
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });

  it('renders a full vocabulary array view widget component', () => {
    const component = renderer.create(
      <ArrayWidget className="metadata">
        {[
          { title: 'Foo', token: 'foo' },
          { title: 'Bar', token: 'bar' },
        ]}
      </ArrayWidget>,
    );
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });
});
