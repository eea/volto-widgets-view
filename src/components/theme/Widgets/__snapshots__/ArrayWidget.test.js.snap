// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArrayWidget renders a full vocabulary array view widget component 1`] = `
<span
  className="metadata array widget"
>
  <span>
    Foo
  </span>
  <span>
    Bar
  </span>
</span>
`;

exports[`ArrayWidget renders a simple array view widget component 1`] = `
<span
  className="metadata array widget"
>
  <span>
    foo
  </span>
  <span>
    bar
  </span>
</span>
`;

exports[`ArrayWidget renders a vocabulary array view widget component 1`] = `
<span
  className="metadata array widget"
>
  <span>
    Foo
  </span>
  <span>
    Bar
  </span>
</span>
`;

exports[`ArrayWidget renders an empty array view widget component 1`] = `
<span
  className="array widget"
>
  
</span>
`;
